metrics/ci_integrations/lib/destination/firestore/config/validation_delegate/firestore_destination_validation_delegate.dart

```diff
@@ -0,0 +1,91 @@
+
+import 'package:ci_integration/client/firestore/models/firebase_auth_credentials.dart';
+import 'package:ci_integration/destination/firestore/config/factory/firebase_auth_factory.dart';
+import 'package:ci_integration/destination/firestore/strings/firestore_strings.dart';
+import 'package:ci_integration/integration/interface/base/config/validation_delegate/validation_delegate.dart';
+import 'package:ci_integration/integration/validation/model/field_validation_result.dart';
+import 'package:firedart/firedart.dart';
+
+/// A [ValidationDelegate] for the Firestore destination integration.
+class FirestoreDestinationValidationDelegate implements ValidationDelegate {
+  /// A [FirebaseAuthFactory] this delegate uses to create [FirebaseAuth]
+  /// instances.
+  final FirebaseAuthFactory authFactory;
+
+  /// Creates a new instance of the [FirestoreDestinationValidationDelegate]
+  /// with the given [authFactory].
+  ///
+  /// Throws an [ArgumentError] if the given [authFactory] is `null`.
+  FirestoreDestinationValidationDelegate(
+    this.authFactory,
+  ) {
+    ArgumentError.checkNotNull(authFactory, 'authFactory');
+  }
+
+  /// Validates the given [firebaseApiKey].
+  Future<FieldValidationResult> validatePublicApiKey(
+    String firebaseApiKey,
+  ) async {
+    try {
+      final auth = authFactory.create(firebaseApiKey);
+
+      await auth.signIn('stub@email.com', 'stub_password');
+    } on FirebaseAuthException catch (e) {
+      final exceptionCode = e.code;
+
+      if (exceptionCode == FirebaseAuthExceptionCode.invalidApiKey) {
+        return const FieldValidationResult.failure(
+          additionalContext: FirestoreStrings.apiKeyInvalid,
+        );
+      }
+    }
+
+    return const FieldValidationResult.success();
+  }
+
+  /// Validates the given Firebase authentication [credentials].
+  Future<FieldValidationResult> validateAuth(
+    FirebaseAuthCredentials credentials,
+  ) async {
+    try {
+      await _createAuth(credentials);
+    } on FirebaseAuthException catch (e) {
+      final exceptionCode = e.code;
+
+      const invalidAuthExceptionCodes = [
+        FirebaseAuthExceptionCode.emailNotFound,
+        FirebaseAuthExceptionCode.invalidPassword,
+        FirebaseAuthExceptionCode.passwordLoginDisabled,
+        FirebaseAuthExceptionCode.userDisabled,
+      ];
+
+      if (invalidAuthExceptionCodes.contains(exceptionCode)) {
+        final message = e.message;
+
+        return FieldValidationResult.failure(additionalContext: message);
+      }
+
+      final code = '${e.code}';
+      final message = e.message;
+      return FieldValidationResult.unknown(
+        additionalContext: FirestoreStrings.authValidationFailedMessage(
+          code,
+          message,
+        ),
+      );
+    }
+
+    return const FieldValidationResult.success();
+  }
+
+  /// Creates a new authenticated [FirebaseAuth] using the given [credentials].
+  Future<FirebaseAuth> _createAuth(FirebaseAuthCredentials credentials) async {
```

Comment:
```suggestion
  Future<FirebaseAuth> _authenticate(FirebaseAuthCredentials credentials) async {
```
Would it be a more descriptive name for this method?